### PR TITLE
CURATOR-241: Write updated data on reconnect

### PR DESCRIPTION
**Pull request by alexbrasetvik**
PersistentEphemeralNode can be initialised with certain data, then later updated. If the client reconnects, the replacing ephemeral should write the updated data.

### CODE REVIEW

File: curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java [hunk 1]
```diff
@@ -338,10 +338,14 @@ public void setData(byte[] data) throws Exception
         this.data.set(Arrays.copyOf(data, data.length));
         if ( isActive() )
         {
-            client.setData().inBackground().forPath(getActualPath(), this.data.get());
+            client.setData().inBackground().forPath(getActualPath(), getData());
         }
     }
 
+    byte[] getData() {
```
Comment (cammckenzie) [inline curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java]: Is there a reason that this is package protected rather than private?

File: curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java [hunk 2]
```diff
@@ -338,10 +338,14 @@ public void setData(byte[] data) throws Exception
         this.data.set(Arrays.copyOf(data, data.length));
         if ( isActive() )
         {
-            client.setData().inBackground().forPath(getActualPath(), this.data.get());
+            client.setData().inBackground().forPath(getActualPath(), getData());
```
Comment (cammckenzie) [inline curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java]: I think that this change isn't required as the data is being accessed directly from within this method.

File: curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java (hunk 1 above)
Comment (alexbrasetvik) [inline curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java]: The idea was to make it available for future tests, but can make it private.

File: curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java (hunk 2 above)
Comment (alexbrasetvik) [inline curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java]: Changed it so it was the same both places actually getting the data, but can revert this change. :)

File: curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java (hunk 1 above)
Comment (cammckenzie) [inline curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java]: I'd mark it as private for now and it can be exposed later if needed. If it were to be for testing we mark these as @VisibleForTesting to be explicit as to why it's package protected.

File: curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java (hunk 2 above)
Comment (cammckenzie) [inline curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java]: Yeah, fair enough. Leave it as is, it's probably nicer to have the data accessed in the same way.

**Comment by alexbrasetvik** [comment]
Updated based on feedback. Want me to squash the commits together? :)

**Comment by cammckenzie** [comment]
Don't worry about squashing the commits. Thanks for the patch.

+1 from me.
